packages/zowe-explorer/src/uss/ZoweUSSNode.ts

```diff
@@ -607,13 +608,34 @@ export class ZoweUSSNode extends ZoweTreeNode implements IZoweUSSTreeNode {
         return path.join(globals.USS_DIR || "", "/" + this.getSessionNode().getProfileName() + "/", this.fullPath);
     }
 
+    public async pasteFileTree(
+        rootPath: string,
+        tree: UssFileTree,
+        uss: {
+            api: ZoweExplorerApi.IUss;
+            options: IUploadOptions;
+        }
+    ) {
+        // Check root path for conflicts before pasting nodes in this path
+        const apiResponse = await uss.api.fileList(rootPath);
+        const fileList = apiResponse.apiResponse?.items;
+        const hasFileOfSameName = fileList.find((file) => file.name === tree.baseName) != null;
+        const fileName = hasFileOfSameName ? `${tree.baseName} (copy)` : tree.baseName;
+        const outputPath = `${rootPath}/${fileName}`;
+
+        await uss.api.fileUtils(outputPath, {
+            request: "copy",
+            from: tree.ussPath,
+            recursive: tree.type === UssFileType.Directory,
+        });
+    }
```

Comment:
@traeok this new api needs to have a check for the existence with extenders ( like is being done in #2086  )  before the call and if doesn't exist ZE should handle the error message instead of expecting each extender to like done in the FTP extension changes above.

Reply:
Thanks for bringing this to my attention @JillieBeanSim - I've pushed some changes such that it will check for `fileUtils` before trying to use that method. If the method exists and the copy/paste operation is within the same session, it will use this new implementation. Otherwise, it will fallback to the original implementation w/ some optimizations.